auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,405 @@
+#  -*- encoding: utf-8 -*-
+##############################################################################
+#     OpenERP, Open Source Management Solution
+#     Copyright (C) 2004-2009 Tiny SPRL (<http://tiny.be>). All Rights Reserved
+#     Copyright 2015 Agile Business Group <http://www.agilebg.com>
+#
+#     This program is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This program is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import xmlrpclib
+import socket
+import os
+import time
+import datetime
+import base64
+import re
+try:
+    import pysftp
+except ImportError:
+    raise ImportError(
+        'This module needs pysftp to automaticly write backups to the FTP '
+        'through SFTP.Please install pysftp on your system.'
+        '(sudo pip install pysftp)'
+    )
```

Comment:
Raise an exception here is a :bug:  Odoo will check that pysftp is present when installing. However you need to catch ImportError to avoid crashing the instance if pysftp is not available and the addon is not installed either.